### PR TITLE
Enable bootstrap.sh to determine OS and version on its own

### PR DESCRIPTION
**Pull request by shishirb-MSFT**

### CODE REVIEW

File: build/scripts/bootstrap.sh [hunk 1]
```diff
@@ -125,82 +111,190 @@ function installBuildDependencies
...
+function determine_machine_architecture()
+{
+    local arch=''
+    arch="$(uname -m)"
+    local ret_val=$?
+    if [[ $ret_val != 0 ]]; then
+        echo "Failed to get cpu architecture."
+        return 1
+    else
+        if [[ $arch == aarch64* || $arch == armv8* ]]; then
```
Comment (JeffMill) [inline build/scripts/bootstrap.sh:196]: see my other comment -- these are mutually exclusive.  e.g. $ARCH_FLAVOR="arm32" would be better IMHO

File: snapcraft-options/snapcraft-agent.yaml [hunk 2]
```diff
@@ -29,7 +29,7 @@ parts:
     plugin: nil
     source: .
     override-build: |
-      ./build/scripts/bootstrap.sh --platform ubuntu2004 --install build
+      ./build/scripts/bootstrap.sh --install build
```
Comment (JeffMill) [inline snapcraft-options/snapcraft-agent.yaml:32]: yay!!!

**Review by JeffMill — APPROVED** [review]
looks good overall, just some suggestions added.

Is there a task on ADU to fix up the way we call DO?

File: build/scripts/bootstrap.sh [hunk 3]
```diff
@@ -125,82 +111,190 @@ function installBuildDependencies
...
-    fi
+    installDeveloperTools
+    installContainerTools
+    installQemu
 }
 
 function isSupportedLinux()
 {
-    if [[ "$PLATFORM" == "debian10" || "$PLATFORM" == "ubuntu1804" || "$PLATFORM" == "ubuntu2004" ]];
+    if [[ ("$OS" == "ubuntu" && ($VER == "18.04" || $VER == "20.04"))
```
[the remark this answers is not in the document]
Comment (shishirb-MSFT) [reply build/scripts/bootstrap.sh:171]: There was no requirement for other versions and we have not tested this on those other versions.

File: build/scripts/bootstrap.sh [hunk 4]
```diff
@@ -125,82 +111,190 @@ function installBuildDependencies
...
-        installDeveloperTools
-        installContainerTools
-        installQemu
-    fi
+    installDeveloperTools
+    installContainerTools
+    installQemu
 }
 
 function isSupportedLinux()
```
[the remark this answers is not in the document]
Comment (shishirb-MSFT) [reply build/scripts/bootstrap.sh:169]: Good point. I'll split it out when there is another script that needs the same functionality.

File: build/scripts/bootstrap.sh (hunk 1 above)
Comment (shishirb-MSFT) [reply build/scripts/bootstrap.sh:196]: Yes, good call.

**Comment by shishirb-MSFT** [comment]
Following up offline.

---
In reply to: [1335530563](https://github.com/microsoft/do-client/pull/162#issuecomment-1335530563) [](http://example.com/codeflow?ancestors=1335530563,1335530563)

**Review by JeffreySaathoff — APPROVED** [review]
:shipit:

**Review by JeffMill — COMMENTED** [review]
awesome, thanks

**Pull request merged**